MAGECLOUD-2739 update services

hguthrie opened the pull request:
This PR updates the **Configure environments** > **Services** section of the Cloud guide.

- updated the versions of services to be general in nature and reference the magento cloud template repository for the latest defaults
- update the service value definitions, specifically the DISK value to be clear about services that require the use of persistent storage
- updated the procedure for enabling the mysql service
- clarified the services connection to Relationships in .magento.app.yaml configuration

whatsnew
-  Updated the [Services topic](https://devdocs.magento.com/guides/v2.2/cloud/project/project-conf-files_services.html) in the Cloud guide to clarify the disk value requirements.
-  Simplified the [relationships property](https://devdocs.magento.com/guides/v2.2/cloud/project/project-conf-files_magento-app.html#relationships) in the application configuration.

NadiyaS commented on guides/v2.2/cloud/project/project-conf-files_services-mysql.md:
```diff
@@ -55,39 +55,42 @@ mysqldb:
             importer:
```
I would also describe that in this case you need to change .magento.app.yaml
So for this example we should have next records in `relationships` section:
`relationships:
    database: "mysqldb:admin"
    databasereporter: "mysqldb:reporter"
    databaseimporter: "mysqldb:importer"`

NadiyaS replied:
and we always should have connection with name `database` with admin rights. You cannot set `database: "mysqldb:reporter"`

NadiyaS replied:
by default, we have `database: "mysql:mysql"` where `mysql` after colon - endpoint with admin permissions - the same as you have `admin in the example above

andriyShevtsov replied:
Also maybe make sense to add link to page about master-slave DB and redis

NadiyaS commented on guides/v2.2/cloud/project/project-conf-files_services-mysql.md:
````diff
@@ -55,39 +55,42 @@ mysqldb:
-```
+  ```yaml
+  relationships:
+      database: "mydatabase:mysql"
````
where this `mydatabase` came from? I see in the example above `mysql` name in services.yaml

hguthrie replied:
That could have been leftover from some old stuff. Sorry to not catch that. (Double-checking my own cloud in case that nonsense came from me.) 😟

NadiyaS commented on guides/v2.2/cloud/project/project-conf-files_services.md:
```diff
@@ -6,93 +6,78 @@ functional_areas:
+### `name`
 
-You can have multiple instances of each service type. For example, you could have multiple Redis instances. For example, we use multiple Redis instances, one for session and one for cache.
+The `name` value identifies the service in the project and consists only of lower case alphanumeric characters: `a` to `z` and `0` to `9`, such as `redis`. You can have multiple instances of each service type. For example, we use multiple Redis instances, one for session and one for cache.
```
I would also add that this `name` is used in `magento.app.yaml` in 
```
relationships:
    database: "<name>:mysql"
```

NadiyaS commented on guides/v2.2/cloud/project/project-conf-files_services.md:
```diff
@@ -6,93 +6,78 @@ functional_areas:
+### `disk`
 
-Specifies the size of the persistent disk storage (in MB) allocated to the service.
+The `disk` value specifies the size of the persistent disk storage (in MB) to allocate to the service. Services that use persistent storage must provide a disk value.
```
actually, there is only one service now which is stored in memory - Redis. and maybe describe it like
```
The `disk` value specifies the size of the persistent disk storage (in MB) to allocate to the service. Every service which uses persistent storage must provide a disk value, when for Redis where all data is in memory, you should not set `disk` field. 
```

andriyShevtsov commented on guides/v2.2/cloud/project/project-conf-files_services-mysql.md:
````diff
@@ -6,40 +6,40 @@ functional_areas:
+Using SSH, log in to the remote server and access the MariaDB database directly:
 
 ```bash
 mysql -h database.internal -u user
````
It will be work only for starter. For PRO customer should take login, password and DB name from RELATIONSHIP

hguthrie commented:
Thank you @NadiyaS  and @andriyShevtsov for the review. I'll update!

hguthrie commented:
I removed a confusing piece about "pretty-print", which was actually a verification step to confirm services were setup properly. I placed an explanation in the main Services topic with procedures for local (using the CLI) and for remote (using PHP)... and then I deleted the include-file about pretty-print. I cleaned all other topics by adding a verification step with a link to the explanation.

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services-mysql.md:
````diff
@@ -55,40 +94,19 @@ mysqldb:
-```
-
-To configure the relationships for the environment variable, set a relationship in your `.magento.app.yaml` in the Git branch. For example:
+The endpoints defined in the above example must also exist in the `relationships` property of the `.magento.app.yaml` file. For example:
````
Not sure:  If you add them here, you have to update the .magento.app.yaml file? --  If that's the case, consider -- If you add additional endpoints, you must also add them to the `relationships` section in the .magento.app.yaml... 

Or verify that the relationships are also included in the ...

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services.md:
```diff
@@ -6,96 +6,139 @@ functional_areas:
-## How this file works {#howitworks}
-
-The `.magento.app.yaml` and `services.yaml` files set the services, applications, and configurations to build and include in an environment. If you add services with specific versions, the initial push and deployment of your branches with these updated files directs the PaaS environment to provision the environment with those services. When you make changes to the services, the environment update ...
+The deploy script uses the configuration files in the `.magento` directory to provision the environment with the configured services. A service becomes available to your application if it is included in the `relationships` property of the `.magento.app.yaml` file. The `services.yaml` file contains the service name, type, and disk values. Changing a service configuration causes a deployment to pro ...
```
```suggestion
The deploy script uses the configuration files in the `.magento` directory to provision the environment with the configured services. A service becomes available to your application if it is included in the `relationships` property of the `.magento.app.yaml` file. The `services.yaml` file contains the service _name_, _type_, and _disk_ values. Changing a service configuration causes a deployment to provision the environment with the updated services.
```

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services.md:
```diff
@@ -6,96 +6,139 @@ functional_areas:
+The `name` value identifies the service in the project. You can only use lower case alphanumeric characters: `a` to `z` and `0` to `9`, such as `redis`.
 
-### `name` {#cloud-yaml-services-name}
+This name value is used in the `relationships` property of the `.magento.app.yaml` configuration file:
```
```suggestion
This _name_ value is used in the `relationships` property of the `.magento.app.yaml` configuration file:
```

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services.md:
````diff
@@ -6,96 +6,139 @@ functional_areas:
 ```
 
-Be aware, if you rename a service in the `services.yaml` file, the following is **permanently removed**:
+Renaming a service in the `services.yaml` file **permanently removes** the following:
````
Maybe make this a warning note since you can potentially lose important data if you miss this.

hguthrie replied:
It is difficult to provide a detailed list with a warning.

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services.md:
````diff
@@ -6,96 +6,139 @@ functional_areas:
+    disk: 2048
+```
+
+The current default storage amount per project is 5GB, or 5120MB. You can distribute this amount between your application and each of its services.
````
Is this per Cloud project, or per environment?  I guess Production and Staging would have same amount if dedicated staging is provisioned.  The current default storage allocated for "persistent data storage" or all types of storage?

hguthrie replied:
This is an old, leftover comment. I did not look in to this and the technical reviewers did not comment.

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services.md:
```diff
@@ -6,96 +6,139 @@ functional_areas:
+## Service relationships
 
-### `disk` {#cloud-yaml-services-disk}
+{{site.data.var.ece}} uses the [`$MAGENTO_CLOUD_RELATIONSHIPS`]({{page.baseurl}}/cloud/env/environment-vars_cloud.html) environment variable to retrieve environment-related relationships. For services to be available to an application in your project, you must specify [*relationships*]({{ page.baseurl }}/cloud/project/project-conf-files_magento-app.html#relationships) between applications and ser ...
```
Verify relationships seems a bit vague -- what am I verifying?

That the updates I made in service.yaml and .magento.app.yaml were applied to the configuration? 
Verifying that all my service relationships are defined correctly?
Verifying that the connections work?  (In the older version, the pretty_print command was introduced by "To print the connection information for Elasticsearch)

meker12 replied:
Adding the additional verbiage introducing the php command, and the extra step at the end of the "verify in remote environment" procedure clarifies nicely :-)

meker12 commented on guides/v2.2/cloud/project/project-conf-files_services.md:
```diff
@@ -6,96 +6,139 @@ functional_areas:
+1.  Use SSH to log in to the remote environment.
 
-## Using the services
+1.  Create `pretty-print` of all relationships for services and configuration data for that environment.
```
After you create the pretty-print, what's next?    Review and test the connections?

hguthrie commented:
running tests
